com.hazelcast.jet.impl.connector.StreamEventJournalP_WmCoalescingTest.when_entryInOnePartition_then_wmForwardedAfterIdleTime

https://hazelcast-l337.ci.cloudbees.com/view/Jet/job/jet-sonar/com.hazelcast.jet$hazelcast-jet-core/1414/testReport/junit/com.hazelcast.jet.impl.connector/StreamEventJournalP_WmCoalescingTest/when_entryInOnePartition_then_wmForwardedAfterIdleTime/

```
Error Message
processor output in mode "snapshots disabled, inboxLimit=1" doesn't match expected:<10
Watermark{ts=02:00:00.010}> but was:<Watermark{ts=02:00:00.010}
10
10
10
Watermark{IDLE_MESSAGE}>
Stacktrace
java.lang.AssertionError: 
processor output in mode "snapshots disabled, inboxLimit=1" doesn't match expected:<10
Watermark{ts=02:00:00.010}> but was:<Watermark{ts=02:00:00.010}
10
10
10
Watermark{IDLE_MESSAGE}>
	at com.hazelcast.jet.core.test.JetAssert.throwNotEqual(JetAssert.java:63)
	at com.hazelcast.jet.core.test.JetAssert.assertEquals(JetAssert.java:46)
	at com.hazelcast.jet.core.test.TestSupport.runTest(TestSupport.java:580)
	at com.hazelcast.jet.core.test.TestSupport.expectOutputs(TestSupport.java:324)
	at com.hazelcast.jet.core.test.TestSupport.expectOutput(TestSupport.java:307)
	at com.hazelcast.jet.impl.connector.StreamEventJournalP_WmCoalescingTest.when_entryInOnePartition_then_wmForwardedAfterIdleTime(StreamEventJournalP_WmCoalescingTest.java:112)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:105)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:97)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/StreamEventJournalP_WmCoalescingTest.java
+++ b/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/connector/StreamEventJournalP_WmCoalescingTest.java
@@ -33,6 +33,8 @@ import org.junit.runner.RunWith;
 
 import java.util.HashSet;
 import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Future;
 import java.util.concurrent.locks.LockSupport;
 import java.util.stream.IntStream;
 
@@ -90,26 +92,31 @@ public class StreamEventJournalP_WmCoalescingTest extends JetTestSupport {
     }
 
     @Test
-    public void when_entryInOnePartition_then_wmForwardedAfterIdleTime() {
+    public void when_entryInOnePartition_then_wmForwardedAfterIdleTime() throws InterruptedException {
         // initially, there will be entries in both partitions
-        map.put(partitionKeys[0], 10);
-        map.put(partitionKeys[1], 10);
+        map.put(partitionKeys[0], 11);
+        map.put(partitionKeys[1], 11);
 
         // insert to map in parallel to verifyProcessor so that the partition0 is not marked as idle
         // but partition1 is
-        new Thread(() -> {
-            for (int i = 0; i < 8; i++) {
+        CountDownLatch productionStartedLatch = new CountDownLatch(1);
+        Future future = spawn(() -> {
+            while (!Thread.interrupted()) {
                 LockSupport.parkNanos(MILLISECONDS.toNanos(500));
-                map.put(partitionKeys[0], 10);
+                map.put(partitionKeys[0], 11);
+                productionStartedLatch.countDown();
             }
-        }).start();
+        });
+        productionStartedLatch.await();
 
-        TestSupport.verifyProcessor(createSupplier(asList(0, 1), 2000))
+        TestSupport.verifyProcessor(createSupplier(asList(0, 1), 4000))
                    .disableProgressAssertion()
                    .disableRunUntilCompleted(4000)
                    .disableSnapshots()
                    .outputChecker((e, a) -> new HashSet<>(e).equals(new HashSet<>(a)))
-                   .expectOutput(asList(10, wm(10)));
+                   .expectOutput(asList(11, wm(11)));
+
+        future.cancel(true);
     }
 
     @Test
@@ -128,7 +135,7 @@ public class StreamEventJournalP_WmCoalescingTest extends JetTestSupport {
             // We will start after a delay so that the source will first become idle and then recover.
             Thread.sleep(4000);
             for (int i = 0; i < 32; i++) {
-                map.put(partitionKeys[0], 10);
+                map.put(partitionKeys[0], 12);
                 Thread.sleep(250);
             }
         }));
@@ -139,10 +146,10 @@ public class StreamEventJournalP_WmCoalescingTest extends JetTestSupport {
                    .disableRunUntilCompleted(8000)
                    .disableSnapshots()
                    .outputChecker((e, a) -> {
-                       a.removeAll(singletonList(10));
+                       a.removeAll(singletonList(12));
                        return a.equals(e);
                    })
-                   .expectOutput(asList(IDLE_MESSAGE, wm(10)));
+                   .expectOutput(asList(IDLE_MESSAGE, wm(12)));
 
         updatingThread.interrupt();
         updatingThread.join();
@@ -151,13 +158,13 @@ public class StreamEventJournalP_WmCoalescingTest extends JetTestSupport {
     @Test
     public void test_nonFirstPartition() {
         /* aim of this test is to check that the mapping from partitionIndex to partitionId works */
-        map.put(partitionKeys[1], 10);
+        map.put(partitionKeys[1], 13);
 
         TestSupport.verifyProcessor(createSupplier(singletonList(1), 2000))
                    .disableProgressAssertion()
                    .disableRunUntilCompleted(4000)
                    .disableSnapshots()
-                   .expectOutput(asList(wm(10), 10, IDLE_MESSAGE));
+                   .expectOutput(asList(wm(13), 13, IDLE_MESSAGE));
     }
 
     public DistributedSupplier<Processor> createSupplier(List<Integer> assignedPartitions, long idleTimeout) {

